WcaOnRails/app/webpacker/components/Results/AttemptResultField/AttemptResultField.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import FmField from './FmField';
+import MbldField from './MbldField';
+import TimeField from './TimeField';
+
+/**
+   `AttemptResultField` is an abstraction over fields specific to the given event,
+   but each of these specific fields works in the similar way.
+
+   The idea behind an attempt result field is that it gets a `value`,
+   allows editing it in any way (by keeping a local draft value)
+   and triggers an `onChange` callback once editing is finished (a blur event).
+
+   This requires keeping both the current value in the parent component (like
+   for a controlled field) and a draft value in the field component
+   (local copy of the upstream value).
+   Whenever `value` changes we want to synchronize the local draft value,
+   which fits into the `getDerivedStateFromProps` lifecycle method.
+   The most straightforward solution would be using useEffect to keep them in sync,
+   but this performs unnecessary re-rendering with old value and leads to jumpy UI.
+   https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html
+   The above article describes alternatives, but none seems good enough in our case:
+   - keeping the draft value in the parent component (e.g. `ResultForm`) is a bad idea,
+   because each field type stores the draft value in different format
+   and it's just best to use the fields as isolated black boxes
+   - using the key prop is a better option, but it remounts the given input element,
+   which may lead to undesired *Tab* key behaviour
+   (clicking *Tab* blurs one input, which may affect the next input value
+   and remounting it as a consequence, in that case we lose focus)
+
+   Using getDerivedStateFromProps sounds justified in this case.
```

Comment:
Ah I didn't make this quite clear, but everything in `AttemptResultField` comes directly from WCA Live!

I just adjusted the UI components used, but otherwise the code hasn't changed, and I'd rather keep it as-is.

However we should make it clear this is coming from there; for `lib/wca-live` I added a readme in the folder, but it doesn't seem like it's enough; what do you think about:
  - moving `AttemptResultField` into a `WCALive` subfolder
  - adding a one-line warning at the top of each file

Reply:
Having a `WCALive` folder within `components` sounds great! Regarding the one-line warning I have no strong preference; I assume having a Readme file within that new `components/WcaLive` folder could well be enough. I'm fine with either solution there.